planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_utils.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/behavior_path_planner_common/utils/utils.hpp"
+
+#include <autoware_test_utils/autoware_test_utils.hpp>
+
+#include <gtest/gtest.h>
+
+using autoware::universe_utils::Point2d;
+using autoware_perception_msgs::msg::PredictedObject;
+using autoware_perception_msgs::msg::PredictedObjects;
+using autoware_planning_msgs::msg::Trajectory;
+using tier4_planning_msgs::msg::PathPointWithLaneId;
+using tier4_planning_msgs::msg::PathWithLaneId;
+using ObjectClassification = autoware_perception_msgs::msg::ObjectClassification;
+
+using autoware::test_utils::createPose;
+using autoware::test_utils::generateTrajectory;
+
+constexpr double epsilon = 1e-6;
+
+PathWithLaneId trajectory_to_path_with_lane_id(const Trajectory & trajectory)
+{
+  PathWithLaneId path_with_lane_id;
+  PathPointWithLaneId path_point_with_lane_id;
+  for (const auto & point : trajectory.points) {
+    path_point_with_lane_id.point.pose = point.pose;
+    path_point_with_lane_id.point.lateral_velocity_mps = point.lateral_velocity_mps;
+    path_point_with_lane_id.point.longitudinal_velocity_mps = point.longitudinal_velocity_mps;
+    path_point_with_lane_id.point.heading_rate_rps = point.heading_rate_rps;
+    path_with_lane_id.points.push_back(path_point_with_lane_id);
+  }
+  return path_with_lane_id;
+}
+
+TEST(BehaviorPathPlanningUtilTest, l2Norm)
+{
+  using autoware::behavior_path_planner::utils::l2Norm;
+
+  geometry_msgs::msg::Vector3 vector = autoware::universe_utils::createVector3(0.0, 0.0, 0.0);
+  auto norm = l2Norm(vector);
+  EXPECT_NEAR(norm, 0.0, epsilon);
+
+  vector = autoware::universe_utils::createVector3(1.0, 2.0, 2.0);
+  norm = l2Norm(vector);
+  EXPECT_NEAR(norm, 3.0, epsilon);
```

Comment:
[nits] I think you can use `EXPECTED_DOUBLE_EQ` as well.
```suggestion
  EXPECT_DOUBLE_EQ(norm, 3.0);
```

Reply:
Fixed in af11169b19a604d54b98a3a988d0474b5fec13ce